core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -109,6 +110,22 @@ public RestStatus status() {
         return getShardInfo().status();
     }
 
+    /**
+     * Gets the location of the written document as a string suitable for a {@code Location} header.
+     * @param routing any routing used in the request. If null the location doesn't include routing information.
+     */
+    public String getLocation(@Nullable String routing) {
+        // Absolute path for the location of the document. This should be allowed as of HTTP/1.1:
```

Comment:
+1, now I see what you're saying (I read that section of the spec. the same way).